library/core/src/ptr/mut_ptr.rs

```diff
@@ -798,11 +801,19 @@ impl<T: ?Sized> *mut T {
     /// unsafe {
     ///     let one = ptr2_other.offset_from(ptr2); // Undefined Behavior! ⚠️
     /// }
-    /// ```
+    /// ```S
     #[stable(feature = "ptr_offset_from", since = "1.47.0")]
     #[rustc_const_stable(feature = "const_ptr_offset_from", since = "1.65.0")]
     #[inline(always)]
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+    #[requires(
+   
+        (mem::size_of::<T>() != 0) &&
+        (self as isize).checked_sub(origin as isize).is_some() &&
+        (self as isize - origin as isize) % (mem::size_of::<T>() as isize) == 0 &&
+        kani::mem::same_allocation(self, origin)
```

Comment:
Can you check if the pointers are pointing to the same address, you can refer to Celina's comments on my [PR](https://github.com/model-checking/verify-rust-std/pull/154)